File: 2843._Count_Symmetric_Integers_E.go
func countSymmetricIntegers(low int, high int) int {
    var result int

    for i := low; i <= high; i++ {
        if i < 100 && i % 11 == 0 {
            result++
        }
        if i > 1000 {
            left  := i / 1000 + i % 1000 / 100
            right := i % 100 / 10 + i % 10
            if  left == right{
                result++
            }
        }
    }

    return result
}

File: 2843._Count_Symmetric_Integers_E.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        result = 0
        for i in range(low, high+1):
            num = str(i)
            len_num = len(num)
            if len_num % 2:
                continue
            sum_left = 0
            sum_right = 0

            for j in range(len_num//2):
                sum_left += int(num[j])
            
            for j in range(len_num//2, len_num):
                sum_right += int(num[j])
            
            if sum_left == sum_right:
                result += 1
        
        return result
